lib/raven/event.rb

```diff
@@ -39,20 +39,21 @@ def initialize(init = {})
       @level         = :error
       @logger        = ''
       @culprit       = nil
-      @server_name   = @configuration.server_name
-      @release       = @configuration.release
-      @modules       = list_gem_specs if @configuration.send_modules
       @user          = {} # TODO: contexts
       @extra         = {} # TODO: contexts
       @server_os     = {} # TODO: contexts
       @runtime       = {} # TODO: contexts
       @tags          = {} # TODO: contexts
       @checksum      = nil
       @fingerprint   = nil
-      @environment   = @configuration.current_environment
 
       yield self if block_given?
 
+      @environment   = @configuration.current_environment
```

Comment:
@nateberkopec I've pushed an alternate implementation that solves my issue without the re-ordering